cmd/host-agent/main.go

```diff
@@ -437,6 +467,34 @@ func main() {
 						ConfigProviderSettings: configProviderSetting,
 					}
 
+					if cfg.AgentFeatures.SyntheticMonitoring {
+						config := worker.Config{
+							Mode:       worker.ModeAgent,
+							Token:      cfg.APIKey,
+							Hostname:   hostname,
+							PulsarHost: cfg.APIURLForSyntheticMonitoring,
+						}
+
+						logger.Info("starting synthetic worker: ")
+						syntheticWorker, err := worker.New(&config)
+						if err != nil {
+							logger.Error("Failed to create worker")
+						}
+
+						go func(ctx context.Context) {
```

Comment:
Let's discuss this block of code. If the intent is to be able to start/stop synthetic checks without restarting the agent,  `ctx` needs to be passed to `syntheticWorker.Run()` so that the following code can return

https://github.com/middleware-labs/synthetics-agent/blob/master/pkg/worker/worker.go#L213